entities/four_year_plan.ts

```diff
@@ -141,9 +141,9 @@ export interface UserSavedSchedule {
   name: string,
   timestamp: number,
   scheduleData: {
-    Major: string,
-    Concentration: string,
-    "Completed Courses": string[],
+    Major: string | undefined,
+    Concentration: string | undefined,
+    "Completed Courses": string[] | undefined,
```

Comment:
Why are these allowed to be undefined, I'm sure there's a good reason but it seems like they shouldn't be.

Reply:
IIRC We had to do that because we were getting some weird error when passing the Schedule Data info from the four year plan page to the Save Schedule Modal, unfortunately I do not remember fully as to why it was needed. @ChrisVasterling May know more as he helped me with that

Reply:
Okay,  if it was from passing it probably is because the interface allows them to be undefined since they are until a user selects them. (Sounds like a good enough reason to me!)